src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/repository/LocationRepository.kt

```diff
@@ -18,3 +20,9 @@ interface LocationRepository : JpaRepository<Location, UUID> {
   @Query("select l from Location l where concat(l.prisonId,'-',l.pathHierarchy) IN (:keys)")
   fun findAllByKeys(keys: List<String>): List<Location>
 }
+
+@Repository
+interface NonResidentialLocationRepository : JpaRepository<NonResidentialLocation, UUID> {
+  @Query("select distinct nrl from NonResidentialLocation nrl, Location l left join nrl.nonResidentialUsages u where u.usageType = :usageType and l.prisonId = :prisonId")
```

Comment:
This doesn't quite right - it doesn't need a left join or an additional Location
```kotlin
@Query("select distinct nrl from NonResidentialLocation nrl join nrl.nonResidentialUsages u where u.usageType = :usageType and l.prisonId = :prisonId")
```

Reply:
refactored